python/taichi/lang/kernel_impl.py

```diff
@@ -574,6 +577,36 @@ def call_back():
                 callbacks.append(get_call_back(v, gpu_v))
         return tmp, callbacks
 
+    def get_paddle_callbacks(self, v, has_pp):
+        callbacks = []
+
+        def get_call_back(u, v):
+            def call_back():
+                u.copy_(v, False)
+
+            return call_back
+
+        assert has_pp
+        assert isinstance(v, paddle.Tensor)
+
+        tmp = v.value().get_tensor()
+        taichi_arch = self.runtime.prog.config.arch
+
+        if str(paddle.device.get_device()).startswith('gpu'):
```

Comment:
Correct me if I were wrong, but I think we should check "Tensor v" 's device instead of the global paddle.device.
You can probably try: `if v.place.is_gpu_place():`

Reply:
> Correct me if I were wrong, but I think we should check "Tensor v" 's device instead of the global paddle.device. You can probably try: `if v.place.is_gpu_place():`

Thanks for your review.  I think you're right.  I didn't call the proper function.